### ISSUE TITLE
Problems with tracking labor cost

### ISSUE THREAD

**Issue by happysalada** [basisproject/tracker#131]
Hi, I have a question regarding tracking labor cost. I wanted the discussion to be available so I thought it might be a good idea to create an issue. Let me know if you think somewhere else would be better.

First of all, I really like the idea of focusing on cost rather than price.
I have trouble imagining the idea when taking into account labor though. It seems to me that when producing something that is 'fairly' straightforward you might have a good idea of how much time is required to do it. However what about things like software? If people are paid by the hour, many people might want to work more on something. If you try to take the criteria that it has to increase value of the final product, then this is also vague. Does making an optimization really improve the final product? 
Same with a book or something more abstract. Did I spend 1 year writing the book or 20 years getting the right skills to write that book? 
The last argument is a little disingenuous I admit, but I think the issue still stands. How can you really define the amount of time that 'should' go into a product.
This has another problem with R&D. Should it be funded? Who decides how much should happen? Is this part of the cost?
The last problem is about how much people should be paid. This one is the hardest. People will tend to want their salaries to be higher, which would increase the costs. Who decides and how do you decide how much 'people' cost? 

In this case, just like you say, capitalism takes the easy way out. Nobody cares how long you spent making this, as long as the price is good enough to generate demand.

**Comment by orthecreedence** [comment]
> Let me know if you think somewhere else would be better.

This is a great place for discussions and questions.

> I have trouble imagining the idea when taking into account labor though. It seems to me that when producing something that is 'fairly' straightforward you might have a good idea of how much time is required to do it. However what about things like software? If people are paid by the hour, many people might want to work more on something. If you try to take the criteria that it has to increase value of the final product, then this is also vague. Does making an optimization really improve the final product?
Same with a book or something more abstract. Did I spend 1 year writing the book or 20 years getting the right skills to write that book?

This is a question I've grappled with myself. I think what we're effectively talking about is the construction of intellectual property. Software, books, art, etc. There are a lot of interesting dynamics at play here.

Let's say you spend a month writing a program, and you "bill" your time at $50/hr (we'll use dollars rather than credits because it's probably easier to mentally work with). Working normal hours, let's say 40 in a week over 4 weeks, that's 160 hours. So you finish and release your program, ignoring any expenses other than your labor, it now has a cost of $50 * 160, or $8000.

Now what? Do you assign costs per-license? You then have to project how many licenses people will want. If you predict more than people want, you'll have uncovered costs. If you predict less than people want, your costs eventually hit zero and then you have a few options:

- stop giving out licenses altogether
- give them out for free
- game the system by logging "labor" but not really doing anything useful

Using a pricing system, this is all somewhat more natural. When basing "price" on cost however, you have to decide up-front how much your labor is worth and hope that what you want is actually useful. This is a lot harder than it might sound. Also, the dynamics change. If most software companies start giving out their software for free after their costs have been realized through licensing, then it might be hard to convince people to initially buy the software when it's released: why not just wait until it's free? To some extent, companies in our current system already deal with this. For instance, blockbuster games release at $60, and as demand drops, the price does as well. Two years after a game is release, the price is often halved.

There's another aspect here that ultimately does change things a lot, and that's UBI. If you get a living wage just for being a member of the system, then maybe instead of $50/hr, you work for $5/hr. Maybe you work for $0/hr. This isn't a silver bullet though. If my services are in high demand, and someone wants me to create IP of some kind (code, a book, etc) then there needs to be some mechanism by which the high cost of my labor can be compensated beyond just "lol u already have an apartment and some baked beans, wat more could u want??"

What you could do is build software via guilds that build many different types of software but bundle accounting in ways such that their products don't have a finite cost per-product as much as the ongoing cost of building and maintaining. Then when you join the guild, you'd negotiate your wage with them and they'd effectively buy your software from you and distribute it at a predetermined cost. Kind of like a publisher I suppose.

Once final aspect is a public company: one with 100% of its costs subsidized that would be enlisted to create IP for effectively the common good. This would be the natural approach to things like infrastructure, medical research, space programs, etc. It could even work for large software companies like search engines where it's effectively a public service with an ongoing cost but you don't necessarily want to charge people per-search to use it. In fact, a search engine is almost like a public library, and those are generally subsidized to all hell.

> This has another problem with R&D. Should it be funded? Who decides how much should happen? Is this part of the cost?

I think I touched on this above, but it's worth talking about on its own I think. I guess the answer might be "it depends." Is the R&D for something a company is producing on its own? I suspect in that case it might sense for them to bundle the cost in with their products. Is the R&D being done for its own purpose? Advancement of scientific knowledge? Public good (medical research)? These types of thing might make sense to fund via pooled public costs (aka taxes), much like they tend to be now.

> In this case, just like you say, capitalism takes the easy way out. Nobody cares how long you spent making this, as long as the price is good enough to generate demand.

It's weird too, because I almost like capitalism better here (*gasp*). If an artist makes something like a cool song, she can make a ton of money from it without knowing (or even trying to know) up front how much her labor is worth. Art is a bit different, because I think most artists would be happy enough to make a living wage UBI and do their art for free, but you get the idea. If I make something that can be copied infinitely, and I do it for a low price, but it actually blows up and people love it, was the cost of my labor really that low wage I initially set? Who knows.

Part of me also believes "IP should be free to everyone." If it can be replicated infinitely, it seems silly to put a price on it! But you also have to allow for compensating people who make IP that others want/need, without compensating those who make IP nobody needs or cares about. Really, if there were an objective way to measure the desire of something before it was even created, that would be the perfect way to solve this. This is impossible though. You can't measure if people like art before it's created. You *can* measure if software is needed to some extent before it's created, but you can't measure if the software itself will adequately meet that need (bad interface, bugs, etc).

This is a difficult and somewhat "unsolved" concept, and I'm glad you posted. I'm open to thoughts and ideas about it. Maybe there are dynamics in play I'm completely missing that could help (or hurt) the Basis way of doing things.

**Comment by happysalada** [comment]
> What you could do is build software via guilds that build many different types of software but bundle accounting in ways such that their products don't have a finite cost per-product as much as the ongoing cost of building and maintaining. Then when you join the guild, you'd negotiate your wage with them and they'd effectively buy your software from you and distribute it at a predetermined cost. Kind of like a publisher I suppose.

I do like the idea of guilds. One simple reason is that I think a lot of the maintenance can be 'painless' if performed by the 'right' people. If there has been an update in a particular library and you have access to a guy who contributes to the library, then chances are he can help a lot. Or on the other end, if somebody has done a particular update for one class of application (a framework or something) then he can probably do the update for a lot more of these. The problem we have with traditional capitalism is that a lot  needs to be re-invented for every closed source software company.

> Once final aspect is a public company: one with 100% of its costs subsidized that would be enlisted to create IP for effectively the common good. This would be the natural approach to things like infrastructure, medical research, space programs, etc. It could even work for large software companies like search engines where it's effectively a public service with an ongoing cost but you don't necessarily want to charge people per-search to use it. In fact, a search engine is almost like a public library, and those are generally subsidized to all hell.

> > This has another problem with R&D. Should it be funded? Who decides how much should happen? Is this part of the cost?
> 
> I think I touched on this above, but it's worth talking about on its own I think. I guess the answer might be "it depends." Is the R&D for something a company is producing on its own? I suspect in that case it might sense for them to bundle the cost in with their products. Is the R&D being done for its own purpose? Advancement of scientific knowledge? Public good (medical research)? These types of thing might make sense to fund via pooled public costs (aka taxes), much like they tend to be now.

my only qualms with this is who gets to decide what gets researched or not. UBI is basically the answer here. As long as everybody gets it, then people are free to do what they want.

> 
> > In this case, just like you say, capitalism takes the easy way out. Nobody cares how long you spent making this, as long as the price is good enough to generate demand.
> 
> It's weird too, because I almost like capitalism better here (_gasp_). If an artist makes something like a cool song, she can make a ton of money from it without knowing (or even trying to know) up front how much her labor is worth. Art is a bit different, because I think most artists would be happy enough to make a living wage UBI and do their art for free, but you get the idea. If I make something that can be copied infinitely, and I do it for a low price, but it actually blows up and people love it, was the cost of my labor really that low wage I initially set? Who knows.
> 
> Part of me also believes "IP should be free to everyone." If it can be replicated infinitely, it seems silly to put a price on it! But you also have to allow for compensating people who make IP that others want/need, without compensating those who make IP nobody needs or cares about. Really, if there were an objective way to measure the desire of something before it was even created, that would be the perfect way to solve this. This is impossible though. You can't measure if people like art before it's created. You _can_ measure if software is needed to some extent before it's created, but you can't measure if the software itself will adequately meet that need (bad interface, bugs, etc).

I can't agree more. There are also interesting network effect that happen when IP is free I believe. New ways to combine things that were previously very hard.

My last question regarding this is how do labor cost get determined? Who says if I should be paid 50$ or 100$?

One potential answer could be a guild. Specialists will decide how much you get paid per hour.

My problem with guilds is that I have a hard time imagining them at scale. 200 people sure. 200 000 though?
Perhaps there could be circles within the guild.

The other caveat I see with having a fixed price per hour is that creative output is vastly different. It means that skilled people accept that even though they are potentially producing 10X more, since this is always hard to evaluate, they are only paid 3X more.

One more caveat is that this requires groups of people to act honestly. What if the guild of builders decides that the guilds of software engineers is paid way too high and so they decide that their work is worth more and they start increasing the price of their labor. Who has the ultimate say on how much a general class of job is paid as a whole?
As an aside I think this is where our capitalist system fails. Giving almost no value to producing food or clothing, always pushing the price of these things down, so that the people producing them have trouble making enough. You know you want to pay food producing roles more, but the problem lies on how much.
In this alternative system https://replacecapitalism.com/chapter-20-a-new-economic-system-the-standard-work-week/
Everybody is paid the same. People can still make more, they just have to work more. That system can still be gamed, since you can't really define 'work' in creative context, so some people might artificially inflate their hours. That system does mostly solve for price per hour problem. Ultimately it's a matter of what people are willing to accept. If there are enough competent people who accept to work for a system and be paid the same as everyone else then it could work (I say competent people, because in that system, they are the ones getting paid less than in the current capitalist system, so they are the one with a tradeoff). I feel it is a big demand considering the status quo, even though I like the idea.

**Comment by orthecreedence** [comment]
Sorry for the late response!

> I do like the idea of guilds. One simple reason is that I think a lot of the maintenance can be 'painless' if performed by the 'right' people. If there has been an update in a particular library and you have access to a guy who contributes to the library, then chances are he can help a lot. Or on the other end, if somebody has done a particular update for one class of application (a framework or something) then he can probably do the update for a lot more of these.

Yeah, it's an interesting model. Sort of a group of independent people (or companies) working towards common goals and sharing resources/accounting. I think it could not only foster a lot of innovation but also abstract away some of the difficulties of running a company solo.

> The problem we have with traditional capitalism is that a lot  needs to be re-invented for every closed source software company.

I agree. There's a lot of wasted effort in competition. I don't necessarily have a problem with competition, however I think capitalism skews heavily towards it and away from cooperation. A system that creates a balance of both will probably yield the most beneficial results. For instance, making it much more common for IP to be shared, but also making it easy to start a competing company/guild if one is stagnating. Ultimately, I view the open source movement as a goal for production: you have many people working together to build things, and sometimes they split projects due to ideological or leadership reasons but ultimately the result is *two great things* instead of *one good thing*. And of course, they often share code back and forth between each other even though they are competing.

> my only qualms with this is who gets to decide what gets researched or not.

My hope would be a bottom-up decision of people who require research. Maybe a collection of farmers or guilds of farmers decide they want to research methods for higher yields while using lower resources. Maybe software companies decide they want research into new languages or frameworks. Maybe a network of hospitals decides they want to research ways to reduce infections.

Right now, most of these things are decided and/or funded by mega-corporations (or the state, who gives corporations a carte-blanche and gets nothing in return), and while a lot of those decisions most likely come from support from the workers inside these corporations, the decisions still need to be ratified by management and those holding the purse strings. I envision a setup where research would flow from people directly doing the labor not just to increase productivity ad nauseum but to make their jobs easier or more fun or more fulfilling.

> UBI is basically the answer here. As long as everybody gets it, then people are free to do what they want.

Yes, this is exactly the idea. You remove the cost of economic survival from the productive system, and what you're left with is an extremely adaptable and flexible network of people producing things because they see the purpose behind them. If you want to research a cure for cancer in your garage, you're free to. It might help to work with a medical center that has others working on the same problem, but that's up to you.


> I can't agree more. There are also interesting network effect that happen when IP is free I believe. New ways to combine things that were previously very hard.

Very true! If you can incentivize sharing of ideas and research instead of charging each other for it, you can make a lot more connections between otherwise seemingly unrelated data.

> My last question regarding this is how do labor cost get determined? Who says if I should be paid 50$ or 100$?
> 
> One potential answer could be a guild. Specialists will decide how much you get paid per hour.
> 
> My problem with guilds is that I have a hard time imagining them at scale. 200 people sure. 200 000 though?
> Perhaps there could be circles within the guild.

I think if you join a guild, they'd likely have guidelines on wages based on experiences and tenure within the guild.

However, Basis itself doesn't have any pre-defined structure for how companies or guilds might work. Effectively, you're free to charge whatever you want for your labor. However, the cost of that labor must be passed on to other companies (or consumers) who use your products/services, and if you want $1M/hr, they might choose not to work with you.

Effectively, it comes down to a labor market. If you cover everyone's basic needs and from there allow them to charge each other based on negotiated wages, you've got a system where people can truly freely associate. If there is a shortage of doctors, the wage will go up and more people will train to become doctors. If there are too many airline pilots, the wage goes down and maybe some of them will re-train into other fields. I believe labor markets are one aspect of capitalism that makes sense. I played with the idea of democratic wages in the past, but ultimately it felt too limiting. Who knows my skill, my training, the job requirements, the stresses/dangers of the job better than me and the people I'm working with? Nobody. So it makes sense we would all negotiate my wage together without involving the general public.

> The other caveat I see with having a fixed price per hour is that creative output is vastly different. It means that skilled people accept that even though they are potentially producing 10X more, since this is always hard to evaluate, they are only paid 3X more.

I agree, hourly wages are not always the best way of doing things. Basis also doesn't define a wage as needing to be hourly. Although the model tracks things in terms of hours, wages themselves can be paid via hourly, salary, bonuses, etc. Companies, guilds, etc are free to set up whatever structure works best for them.

> One more caveat is that this requires groups of people to act honestly. What if the guild of builders decides that the guilds of software engineers is paid way too high and so they decide that their work is worth more and they start increasing the price of their labor. Who has the ultimate say on how much a general class of job is paid as a whole?

I suppose ultimately, like capitalism, it would be the consumers of their services. While there's some disconnect in Basis between the cost (in labor/resources) of some product and the price it sells for when it's consumed, in general the price follows the cost. So if the cost of labor goes too high, less people consume it, and it puts downward pressure on the cost of that product (and the cost of the labor or resources behind it).

One interesting thought I've had lately is to fix wages to be some value between 0 and 1. This forces people to think of wages as what they truly are: a comparative differential between separate jobs. 1 would mean "low supply, high demand" and 0 would be "high supply, low demand."

That said, this would force wages to be one particular type (hourly, salary, etc) across the board which is undesirable and inflexible. Probably won't happen, but interesting to think about!

> As an aside I think this is where our capitalist system fails. Giving almost no value to producing food or clothing, always pushing the price of these things down, so that the people producing them have trouble making enough. You know you want to pay food producing roles more, but the problem lies on how much.

Interestingly, I think living wage UBI turns some of this on its head. Working in textiles or food is really hard work. So if you already have a living wage, would you do it? Many who are pushed into these positions *now* might choose not to do them if they have a UBI. So what's the solve here? Raise the wage until someone is willing to do the work!

I think we'd find that difficult, demanding, stressful jobs (even if "unskilled") would start fetching higher wages simply because you have to make it worth someone's while if they have their basic needs met.

> In this alternative system https://replacecapitalism.com/chapter-20-a-new-economic-system-the-standard-work-week/
> Everybody is paid the same. People can still make more, they just have to work more. That system can still be gamed, since you can't really define 'work' in creative context, so some people might artificially inflate their hours. That system does mostly solve for price per hour problem. Ultimately it's a matter of what people are willing to accept. If there are enough competent people who accept to work for a system and be paid the same as everyone else then it could work (I say competent people, because in that system, they are the ones getting paid less than in the current capitalist system, so they are the one with a tradeoff). I feel it is a big demand considering the status quo, even though I like the idea.

Yeah I'm really often enticed by the idea of everyone getting paid the same (or not getting paid at all). However, I've been over it so many times in my head and while I don't think it's against "human nature" (which is such a terrible argument because human nature is extremely expansive and adaptable) I think people will get jealous. Why does someone who works a much easier job than me get paid the same? Then you get into things like rotating jobs and all that to make things "fair." I could see this working possibly, but in itself would take a lot of organizational work and I wonder what problem is really being solved.

In other words, I'm conflicted on the idea. I'd love to see a world without wages, where people just take what they need. I've advocated for such a world many times in the past. I just don't know how it would actually work, at least in humanity's current stage of development.

Ultimately, there's going to be a large reconciling soon between the number of people on the planet and the number of finite resources on the planet. I'm unconvinced this reconciliation can happen without some form of numeric currency.

**Comment by happysalada** [comment]
One more point that came to me was around the asymetry of information in capitalism. Currently, we buy a lot of crap since we can't really differentiate between what is really good and what is not. This encourages people to charge higher to compensate for advertising cost. Another way to solve this that came to me recently was around having 'buying guilds'. Or experts getting together to define what is good and what is not. You don't see this very often in the modern world, people getting together to define what is worthwhile. You see influencers who are paid, so biased,, but I was thinking that if there were a group of people, they would be corrupted less easily. Groups of people have stronger bargaining power. The one problem with buying guilds is that they need to be incentivized not on sales, otherwise it's the old system again. I keep on thinking that if there is stronger opposition to so many companies producing so many things, we could get closer to a price that reflects costs rather than a price subsidizing advertising.
 
All of these ideas require cooperation, which is nothing new. Sometimes I wonder how we can trigger a shift in mindset? Do we need to create tools? My answer at the moment is that since it's a mindset problem, we can't do much except wait until people are ready. Publications of an alternative view of the world, might be the best thing that gets us into a cooperative mindset.

**Comment by orthecreedence** [comment]
> Currently, we buy a lot of crap since we can't really differentiate between what is really good and what is not. This encourages people to charge higher to compensate for advertising cost. Another way to solve this that came to me recently was around having 'buying guilds'.

This is a really cool idea, and could even be a joint consumer-producer co-op. If the first round of production sucks, the consumers would have some say in how the product is made better. This definitely speaks to your point about information asymmetry...people in capitalism really can only say "this product is trash" by not buying it again. Why is it trash? What things didn't work for you? Companies make it really difficult to call them up and tell them what happened. Having some kind of participatory structure that uses direct communication rather than just market mechanisms would greatly improve product longevity and cut down on resource usage by magnitudes.

I think having any kind of retail outlet have the option for some degree of consumer management would be great. It kind of drives the point home that "people have influence in the things affecting them." I'll think about how this would fit into the protocol.

> All of these ideas require cooperation, which is nothing new. Sometimes I wonder how we can trigger a shift in mindset? Do we need to create tools? My answer at the moment is that since it's a mindset problem, we can't do much except wait until people are ready. Publications of an alternative view of the world, might be the best thing that gets us into a cooperative mindset.

Yes, it's very much a cultural problem. It's not that nobody has done it before, but moreso it's uncommon enough that people just don't think to do it. Beyond that, the economics are difficult. Our financial and consumer system is set up to favor private ownership and shoddy craftsmanship. Who makes more money: they company that makes washing machines that last 30 years and have easily-replaceable parts, or the company that pumps our washing machines that break every two years? Our system tips the scales in favor of planned obsolescence. Maybe there needs to be a unified push of cooperation *and* the economic structure to support it.

And that brings up a great question: lets say we make a great post-capitalist protocol that would produce better products, distribute them to more people, and make everyone happier...how do you get people to use it? It's really a cultural shift that would have to happen slowly. I have lots of ideas for this transition, but there's not really one killer idea that makes it happen. It will likely be a long, difficult path. What we're building is a network, and that takes a lot of time and effort *especially* in the beginning.

**Comment by happysalada** [comment]
> And that brings up a great question: lets say we make a great post-capitalist protocol that would produce better products, distribute them to more people, and make everyone happier...how do you get people to use it? It's really a cultural shift that would have to happen slowly. I have lots of ideas for this transition, but there's not really one killer idea that makes it happen. It will likely be a long, difficult path. What we're building is a network, and that takes a lot of time and effort _especially_ in the beginning.

This made me think: What would be the appropriate financial incentive for such a structure to happen? (if it's even a problem of financial incentive). 
For physical products
- Consummers pay a little bit every year until the product is returned (because broken or don't want to use it anymore). I like this idea because companies are responsible for recycling what they produce. It could be a good thing because companies might see more profit in this way (if you manage to make your product last longer than the estimated time for example. Say you think this computer will last 5 years, if it costs 500$ you charge 100$ per year and could start turning a profit after year 5). I don't like this concept of extracting money form the consumer, but it is an incentive for current capitalist companies to change their business model.
    This basically sounds like some form of warranty to me, and those already exists. So perhaps it's more of a mindset problem rather than a financial problem. Consumers don't have the knowledge to compare things properly, so even if a product gives you a 5 year warranty, you still hesitate with one giving you a 1 year warranty.

> This is a really cool idea, and could even be a joint consumer-producer co-op. If the first round of production sucks, the consumers would have some say in how the product is made better. This definitely speaks to your point about information asymmetry...people in capitalism really can only say "this product is trash" by not buying it again. Why is it trash? What things didn't work for you? Companies make it really difficult to call them up and tell them what happened. Having some kind of participatory structure that uses direct communication rather than just market mechanisms would greatly improve product longevity and cut down on resource usage by magnitudes.

This made me think about a service to companies where they can "handle" all the user feedback. When I say service, I'm just thinking of a practical way to enable that transition. I have the feeling at the moment, companies don't include customers in their decisions because it's hard and costly to handle feedback of many people. The most companies are involved is through social media. The main use of social media to me is more of a "shaming" the company when they do something wrong. The problem with building anything that groups consumer feedback is that ultimately it will be a time costly process, probably most companies do not want to spend the time to do this properly unless there are proofs that it generates more profits.

Another thought that came to mind is that communities of passionate people exist already, it's basically reddit. However reddit doesn't really have a way to properly organize knowledge (last I checked). So it's just an infinite scroll with new content all the time. I wonder if communities would find it valuable to have a wiki where they could store information about something so that other people could check it. I guess those "best 10 products of 2021" that are provided by media would have a lot more weight if actual people could vote on those and provide their review. Maybe it's just a review site then. The problem with a review site is that you have to make it worthwhile for the initial people to post reviews. Perhaps it's just reddit with a wiki feature then. I guess it's about understanding what tools existing communities would use to organize better.
